### PR TITLE
Add error computation for partitioned-heat-conduction OpenFOAM solver

### PR DESCRIPTION
**Pull request by davidscn**
To help testing and comparing.

Related to https://github.com/precice/tutorials/issues/445

In our tutorials, with OpenFOAM - OpenFOAM, I get

Dirichlet:
```
Error metrics at t = 1s:
l2 error (sqrt(sum(err^2)/n)):		6.2733e-05
Maximum absolute error (max(err)):	7.72464e-05
Global absolute error (sum(err^2)):	3.93543e-05
```

Neumann
```
Error metrics at t = 1s:
l2 error (sqrt(sum(err^2)/n)):		6.24377e-05
Maximum absolute error (max(err)):	7.4964e-05
Global absolute error (sum(err^2)):	3.89846e-05
```

### CODE REVIEW

**Review by MakisH — APPROVED** [review]
This looks like something we can easily merge today.

Apart from the value of `beta` (0.2 instead of 0.3, for consistency with the rest of the tutorials), this is identical to the OpenFOAM-preCICE paper code.

**Comment by MakisH** [comment]
I merged develop into this branch. The diff looks a bit different just because of the new formatting config added in #468.

File: partitioned-heat-conduction/solver-openfoam/heatTransfer.C [hunk 1]
```diff
@@ -61,8 +101,8 @@ int main(int argc, char *argv[])
 
   const double alpha = 3;
   const double beta  = 1.2;
```
Comment (BenjaminRodenberg) [inline partitioned-heat-conduction/solver-openfoam/heatTransfer.C:103]: @MakisH to help with archeology here: In https://github.com/precice/tutorials/pull/379/ I changed everything to 1.2 according to the FEniCS book.

Comment (MakisH) [reply partitioned-heat-conduction/solver-openfoam/heatTransfer.C:103]: ok, then the OpenFOAM-preCICE publication is still in synch with the old tutorials, and the new tutorials are in sync with each other. All good, thanks!

**Pull request merged**